kubectl-cloudflow/verify/verifiedblueprint.go

```diff
@@ -0,0 +1,272 @@
+}
+
+type VerifiedInlet struct {
+	VerifiedPort
+	streamlet VerifiedStreamlet
+}
+
+type VerifiedOutlet struct {
+	VerifiedPort
+	streamlet VerifiedStreamlet
+}
+
+type VerifiedStreamletConnection struct {
+	verifiedOutlet VerifiedOutlet
+	verifiedInlet  VerifiedInlet
+	label          *string
+}
+
+type VerifiedBlueprint struct {
+	streamlets  []VerifiedStreamlet
+	connections []VerifiedStreamletConnection
+}
+
+func (v *VerifiedStreamlet) outlet(outlet OutletDescriptor) VerifiedOutlet {
+	return VerifiedOutlet{
+		VerifiedPort{
+			outlet.Name,
+			outlet.Schema,
+		},
+		*v,
+	}
+}
+
+func (v *VerifiedStreamlet) inlet(inlet InletDescriptor) VerifiedOutlet {
+	return VerifiedOutlet{
+		VerifiedPort{
+			inlet.Name,
+			inlet.Schema,
+		},
+		*v,
+	}
+}
+
+// ToString string representation of a VerifiedPortPath
+func (v VerifiedPortPath) ToString() string {
+	if v.portName == nil {
+		return v.streamletRef
+	}
+	return fmt.Sprintf("%s.%s", v.streamletRef, *v.portName)
+}
+
+// NewVerifiedPortPath constructs a VerifiedPortPath out of a string
+func NewVerifiedPortPath(portPath string) (*VerifiedPortPath, *InvalidPortPath) {
+	var trimmed = strings.TrimSpace(portPath)
+	var splitF = func(c rune) bool {
+		return c == '.'
+	}
+	var parts = strings.FieldsFunc(trimmed, splitF)
+
+	if strings.HasPrefix(trimmed, ".") {
+		return nil, &InvalidPortPath{
+			PortPathError: PortPathError{},
+			path:          portPath}
+	} else if len(parts) == 1 && !strings.HasSuffix(portPath, ".") {
+		return &VerifiedPortPath{parts[0], nil}, nil
+	} else if len(parts) >= 2 {
+		var portName = parts[len(parts)-1]
+		var streamletNamePart = parts[:len(parts)-1]
+		var streamletRef = mkString(streamletNamePart, ".")
+		if len(streamletRef) == 0. {
+			return nil, &InvalidPortPath{
+				PortPathError: PortPathError{},
+				path:          portPath}
+		} else if len(portName) == 0 {
+			return nil, &InvalidPortPath{
+				PortPathError: PortPathError{},
+				path:          portPath}
+		} else {
+			return &VerifiedPortPath{parts[0], &parts[1]}, nil
+		}
+	} else {
+		return nil, &InvalidPortPath{
+			PortPathError: PortPathError{},
+			path:          portPath}
+	}
+}
+
+func (v *VerifiedInlet) portPath() VerifiedPortPath {
+	return VerifiedPortPath{
+		v.streamlet.name,
+		&v.VerifiedPort.portName,
+	}
+}
+
+func (v *VerifiedOutlet) portPath() VerifiedPortPath {
+	return VerifiedPortPath{
+		v.streamlet.name,
+		&v.VerifiedPort.portName,
+	}
+}
+
+func (v *VerifiedOutlet) matches(outletDescriptor OutletDescriptor) bool {
+	return outletDescriptor.Name == v.portName && outletDescriptor.Schema.Fingerprint == v.schemaDescriptor.Fingerprint
+}
+
+// FindVerifiedOutlet finds the VerifiedOutlet corresponding to the outletPortPath from the list of
+// verifiedStreamlets
+func FindVerifiedOutlet(verifiedStreamlets []VerifiedStreamlet, outletPortPath string) (*VerifiedOutlet, BlueprintProblem) {
+	verifiedPortPath, err := NewVerifiedPortPath(outletPortPath)
```

Comment:
Could move this down a line and leave the space between the function header and the function for redability

Reply:
afaik this is how go libs document stuff (https://blog.golang.org/godoc-documenting-go-code).
```
The convention is simple: to document a type, variable, constant, function, or even a package, write a regular comment directly preceding its declaration, with no intervening blank line. ```

Reply:
ok sure but we have to align the whole code base, this is part of improving the code using a well-defined style guide.